ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -452,4 +466,62 @@ public void shouldSetPropertyInRunScript() {
     assertThat(overriddenProperties.get("auto.offset.reset"), equalTo("earliest"));
 
   }
+
```

Comment:
We should add a test that verifies that new queries can be started if a terminate brings the active count below the threshold.